independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -71,76 +91,141 @@ static BeanInfo createClassBean(ClassInfo beanClass, BeanDeployment beanDeployme
+            }
+            if (name == null) {
+                name = initStereotypeName(stereotypes, beanClass);
+            }
+
+            if (isAlternative) {
+                alternativePriority = initAlternativePriority(beanClass, alternativePriority, stereotypes, beanDeployment);
+                if (alternativePriority == null) {
+                    // after all attempts, priority is still null, bean will be ignored
+                    LOGGER.debugf(
+                            "Ignoring bean defined via %s - declared as an @Alternative but not selected by @Priority, @AlternativePriority or quarkus.arc.selected-alternatives",
+                            beanClass.name());
+                    return null;
+                }
+            }
+
+            return new BeanInfo(beanClass, beanDeployment, scope, types, qualifiers,
+                    Injection.forBean(beanClass, null, beanDeployment, transformer), null, null,
+                    isAlternative ? alternativePriority : null, stereotypes, name, isDefaultBean);
         }
-        if (name == null) {
-            name = initStereotypeName(stereotypes, beanClass);
+
+        private static void processSuperClass(ClassInfo beanClass, BeanDeployment beanDeployment,
+                Set<AnnotationInstance> qualifiers, List<StereotypeInfo> stereotypes) {
+            DotName superClassName = beanClass.superName();
+            while (!superClassName.equals(DotNames.OBJECT)) {
+                ClassInfo classFromIndex = getClassByName(beanDeployment.getBeanArchiveIndex(), superClassName);
+                if (classFromIndex == null) {
+                    // class not in index
+                    LOGGER.warnf("Unable to get inherited qualifier of stereotype for bean %s because its " +
+                            "super class %s is not part of Jandex index. This inheritance will not apply.",
+                            beanClass,
+                            superClassName);
+                    break;
+                }
+                for (AnnotationInstance annotation : beanDeployment.getAnnotationStore().getAnnotations(classFromIndex)) {
+                    processInheritedAnnotation(annotation, beanDeployment, qualifiers, stereotypes);
+                }
+                superClassName = classFromIndex.superName();
+            }
         }
 
-        if (isAlternative) {
-            alternativePriority = initAlternativePriority(beanClass, alternativePriority, stereotypes, beanDeployment);
-            if (alternativePriority == null) {
-                // after all attempts, priority is still null, bean will be ignored
-                LOGGER.debugf(
-                        "Ignoring bean defined via %s - declared as an @Alternative but not selected by @Priority, @AlternativePriority or quarkus.arc.selected-alternatives",
-                        beanClass.name());
-                return null;
+        private static void processInterface(ClassInfo classInfo,
```

Comment:
CDI doesn't inherit metadata from interfaces, only subclasses. See https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#inheritance

Reply:
holly crap, I do that part for nothing ;-). I will remove it.